common/dal2/tests/it/services/s3.rs

```diff
@@ -26,6 +27,34 @@ async fn builder() {
         .credential(Credential::hmac("access-key", "secret-key"))
         .endpoint("http://localhost:9000")
         .finish()
+        .unwrap();
+}
+
+#[tokio::test]
+// TODO: we need to add an integration test for it.
```

Comment:
Does this test will be covered by our statefultest?
https://github.com/datafuselabs/databend/blob/a479e82a5285645393eec05707ea2fe978b9e28e/.github/actions/test_stateful_standalone/action.yml#L52

Reply:
> Maybe we should impl a mock to service the unit test?

I don't like the idea to mock storage services. In my experience, most of the mock behave inconsistently with the real one.

I prefer to test on a real storage service like local fs, minio.